systemtest: TestProfiling is flaky

This test seems to uncover some flakiness with gRPC setup and the profiling test and is preventing succesful systemtests in `main` and `8.5`. 

```
=== RUN   TestProfiling
2022/10/04 01:04:53 Starting container id: 71cb771199c4 image: elastic-agent-systemtest:8.6.0-158a13db-SNAPSHOT
2022/10/04 01:04:53 Waiting for container id 71cb771199c4 image: elastic-agent-systemtest:8.6.0-158a13db-SNAPSHOT
2022/10/04 01:05:01 Container is ready id: 71cb771199c4 image: elastic-agent-systemtest:8.6.0-158a13db-SNAPSHOT
    profiling_test.go:120: 
        	Error Trace:	/var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9248/src/github.com/elastic/apm-server/systemtest/profiling_test.go:120
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unavailable desc = connection closed before server preface received
        	Test:       	TestProfiling
    fleet_test.go:179: elastic-agent logs: {"log.level":"info","@timestamp":"2022-10-04T01:04:55.052Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":471},"message":"Starting enrollment to URL: [https://fleet-server:8220/","ecs.version":"1.6.0"}](https://fleet-server:8220/)
        {"log.level":"info","@timestamp":"2022-10-04T01:04:55.861Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":271},"message":"Elastic Agent might not be running; unable to trigger restart","ecs.version":"1.6.0"}
        Successfully enrolled the Elastic Agent.
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.019Z","log.origin":{"file.name":"cmd/run.go","file.line":202},"message":"APM instrumentation disabled","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.021Z","log.origin":{"file.name":"application/application.go","file.line":78},"message":"Detecting execution mode","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.039Z","log.origin":{"file.name":"application/application.go","file.line":102},"message":"Agent is managed by Fleet","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.039Z","log.origin":{"file.name":"capabilities/capabilities.go","file.line":59},"message":"capabilities file not found in /usr/share/elastic-agent/state/capabilities.yml","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.367Z","log.logger":"composable.providers.docker","log.origin":{"file.name":"docker/docker.go","file.line":44},"message":"Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.368Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":68},"message":"Starting stats endpoint","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.368Z","log.origin":{"file.name":"application/managed_mode.go","file.line":316},"message":"Agent is starting","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:56.368Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":70},"message":"Metrics endpoint listening on: /usr/share/elastic-agent/state/data/tmp/elastic-agent.sock (configured: unix:///usr/share/elastic-agent/state/data/tmp/elastic-agent.sock)","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:58.206Z","log.origin":{"file.name":"artifact/config.go","file.line":138},"message":"Source URI changed from \"[https://artifacts.elastic.co/downloads/\"](https://artifacts.elastic.co/downloads//) to \"[https://artifacts.elastic.co/downloads/\"","ecs.version":"1.6.0"}](https://artifacts.elastic.co/downloads//)
        {"log.level":"info","@timestamp":"2022-10-04T01:04:58.292Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":48},"message":"New State ID is zQrdalpX","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:58.292Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":49},"message":"Converging state requires execution of 2 step(s)","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:58.441Z","log.origin":{"file.name":"operation/operator.go","file.line":387},"message":"operation 'operation-install' skipped for apm-server.8.6.0-SNAPSHOT","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:58.656Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-10-04T01:04:58Z - message: Application: apm-server--8.6.0-SNAPSHOT[cdec6d3e-d8cc-4e52-9620-342b937cfff8]: State changed to STARTING: Starting - type: 'STATE' - sub_type: 'STARTING'","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:58.985Z","log.origin":{"file.name":"operation/operator.go","file.line":387},"message":"operation 'operation-install' skipped for filebeat.8.6.0-SNAPSHOT","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:59.512Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-10-04T01:04:59Z - message: Application: filebeat--8.6.0-SNAPSHOT--36643631373035623733363936343635[cdec6d3e-d8cc-4e52-9620-342b937cfff8]: State changed to STARTING: Starting - type: 'STATE' - sub_type: 'STARTING'","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:59.718Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-10-04T01:04:59Z - message: Application: apm-server--8.6.0-SNAPSHOT[cdec6d3e-d8cc-4e52-9620-342b937cfff8]: State changed to CONFIG: Updating configuration - type: 'STATE' - sub_type: 'CONFIG'","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:04:59.948Z","log.origin":{"file.name":"operation/operator.go","file.line":387},"message":"operation 'operation-install' skipped for metricbeat.8.6.0-SNAPSHOT","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:05:00.395Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-10-04T01:05:00Z - message: Application: metricbeat--8.6.0-SNAPSHOT--36643631373035623733363936343635[cdec6d3e-d8cc-4e52-9620-342b937cfff8]: State changed to STARTING: Starting - type: 'STATE' - sub_type: 'STARTING'","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:05:00.406Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":66},"message":"Updating internal state","ecs.version":"1.6.0"}
        {"log.level":"info","@timestamp":"2022-10-04T01:05:00.757Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-10-04T01:05:00Z - message: Application: filebeat--8.6.0-SNAPSHOT--36643631373035623733363936343635[cdec6d3e-d8cc-4e52-9620-342b937cfff8]: State changed to RUNNING: Running - type: 'STATE' - sub_type: 'RUNNING'","ecs.version":"1.6.0"}
    fleet_test.go:181: apm-server logs:
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.689Z","log.origin":{"file.name":"beatcmd/beat.go","file.line":155},"message":"Beat ID: 727047a8-bb33-410f-acb2-d58de13e8bf4","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.690Z","log.origin":{"file.name":"beatcmd/beat.go","file.line":168},"message":"Set gc percentage to: 100","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.690Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":63},"message":"Starting stats endpoint","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.691Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":65},"message":"Metrics endpoint listening on: /usr/share/elastic-agent/state/data/tmp/default/apm-server/apm-server.sock (configured: unix:///usr/share/elastic-agent/state/data/tmp/default/apm-server/apm-server.sock)","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.690Z","log.logger":"beat","log.origin":{"file.name":"beatcmd/beat.go","file.line":605},"message":"Beat info","service.name":"apm-server","system_info":{"beat":{"path":{"config":"/usr/share/elastic-agent/data/elastic-agent-e18405/install/apm-server-8.6.0-SNAPSHOT-linux-x86_64","data":"/usr/share/elastic-agent/state/data/run/default/apm-server--8.6.0-SNAPSHOT","home":"/usr/share/elastic-agent/data/elastic-agent-e18405/install/apm-server-8.6.0-SNAPSHOT-linux-x86_64","logs":"/usr/share/elastic-agent/data/elastic-agent-e18405/install/apm-server-8.6.0-SNAPSHOT-linux-x86_64/logs"},"type":"apm-server","uuid":"727047a8-bb33-410f-acb2-d58de13e8bf4"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.692Z","log.logger":"beat","log.origin":{"file.name":"beatcmd/beat.go","file.line":613},"message":"Build info","service.name":"apm-server","system_info":{"build":{"commit":"0a8e0ab7da29b7a617ad3bfa5c27ec1a29e3665e","time":"2022-10-03T10:41:43.000Z","version":"8.6.0"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.692Z","log.logger":"beat","log.origin":{"file.name":"beatcmd/beat.go","file.line":616},"message":"Go runtime info","service.name":"apm-server","system_info":{"go":{"os":"linux","arch":"amd64","max_procs":4,"version":"go1.18.6"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.693Z","log.logger":"beat","log.origin":{"file.name":"beatcmd/beat.go","file.line":620},"message":"Host info","service.name":"apm-server","system_info":{"host":{"architecture":"x86_64","boot_time":"2022-10-04T00:50:59Z","containerized":true,"name":"71cb771199c4","ip":["127.0.0.1/8","172.18.0.9/16"],"kernel_version":"5.4.0-1087-gcp","mac":["02:42:ac:12:00:09"],"os":{"type":"linux","family":"debian","platform":"ubuntu","name":"Ubuntu","version":"20.04.5 LTS (Focal Fossa)","major":20,"minor":4,"patch":5,"codename":"focal"},"timezone":"UTC","timezone_offset_sec":0},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.693Z","log.logger":"beat","log.origin":{"file.name":"beatcmd/beat.go","file.line":649},"message":"Process info","service.name":"apm-server","system_info":{"process":{"capabilities":{"inheritable":null,"permitted":null,"effective":null,"bounding":["chown","dac_override","fowner","fsetid","kill","setgid","setuid","setpcap","net_bind_service","net_raw","sys_chroot","mknod","audit_write","setfcap"],"ambient":null},"cwd":"/usr/share/elastic-agent/data/elastic-agent-e18405/install/apm-server-8.6.0-SNAPSHOT-linux-x86_64","exe":"/usr/share/elastic-agent/data/elastic-agent-e18405/install/apm-server-8.6.0-SNAPSHOT-linux-x86_64/apm-server","name":"apm-server","pid":27,"ppid":7,"seccomp":{"mode":"filter","no_new_privs":false},"start_time":"2022-10-04T01:04:57.820Z"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.703Z","log.logger":"esclientleg","log.origin":{"file.name":"eslegclient/connection.go","file.line":108},"message":"elasticsearch url: [http://localhost:9200","service.name":"apm-server","ecs.version":"1.6.0"}](http://localhost:9200/)
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.703Z","log.logger":"publisher","log.origin":{"file.name":"pipeline/module.go","file.line":113},"message":"Beat name: 71cb771199c4","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.704Z","log.origin":{"file.name":"beatcmd/beat.go","file.line":397},"message":"apm-server started.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2022-10-04T01:04:58.704Z","log.logger":"cfgwarn","log.origin":{"file.name":"management/manager.go","file.line":116},"message":"BETA: Fleet management is enabled","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.704Z","log.logger":"centralmgmt.fleet","log.origin":{"file.name":"management/manager.go","file.line":117},"message":"Starting fleet management service","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.704Z","log.logger":"centralmgmt.fleet","log.origin":{"file.name":"management/manager.go","file.line":125},"message":"Ready to receive configuration","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:58.705Z","log.logger":"beater","log.origin":{"file.name":"beater/beater.go","file.line":1161},"message":"maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.219Z","log.logger":"centralmgmt.fleet","log.origin":{"file.name":"management/manager.go","file.line":162},"message":"Status change to Configuring: Updating configuration","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.221Z","log.logger":"centralmgmt.fleet","log.origin":{"file.name":"management/manager.go","file.line":287},"message":"Applying settings for inputs","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2022-10-04T01:04:59.222Z","log.logger":"cfgwarn","log.origin":{"file.name":"tlscommon/config.go","file.line":102},"message":"DEPRECATED: Treating the CommonName field on X.509 certificates as a host name when no Subject Alternative Names are present is going to be removed. Please update your certificates if needed. Will be removed in version: 8.0.0","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.223Z","log.logger":"config","log.origin":{"file.name":"config/rum.go","file.line":86},"message":"Unable to determine sourcemap storage, sourcemaps will not be applied","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.226Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":142},"message":"Listening on: [::]:8200","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.226Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":71},"message":"creating transaction metrics aggregation with config: {Interval:1m0s MaxTransactionGroups:10000 HDRHistogramSignificantFigures:2}","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.236Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":86},"message":"creating service destinations aggregation with config: {Interval:1m0s MaxGroups:10000}","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.236Z","log.logger":"centralmgmt.fleet","log.origin":{"file.name":"management/manager.go","file.line":287},"message":"Applying settings for output","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.238Z","log.logger":"config","log.origin":{"file.name":"config/rum.go","file.line":89},"message":"Falling back to elasticsearch output for sourcemap storage","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.239Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":142},"message":"Listening on: [::]:8200","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.240Z","log.logger":"beater","log.origin":{"file.name":"beater/waitready.go","file.line":40},"message":"blocking ingestion until all preconditions are satisfied","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.243Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":71},"message":"creating transaction metrics aggregation with config: {Interval:1m0s MaxTransactionGroups:10000 HDRHistogramSignificantFigures:2}","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.257Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":86},"message":"creating service destinations aggregation with config: {Interval:1m0s MaxGroups:10000}","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.259Z","log.logger":"beater","log.origin":{"file.name":"beater/waitready.go","file.line":68},"message":"no longer blocking ingestion as all precondition checks are now satisfied","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":348},"message":"registered profiling collection (technical preview)","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path / added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /config/v1/agents added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /config/v1/rum/agents added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /intake/v2/rum/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /intake/v3/rum/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /intake/v2/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /v1/traces added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /v1/metrics added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /v1/logs added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.324Z","log.logger":"beater","log.origin":{"file.name":"beater/server.go","file.line":193},"message":"Starting apm-server [0a8e0ab7da29b7a617ad3bfa5c27ec1a29e3665e built 2022-10-03 12:41:43 +0200 +0200]. Hit CTRL-C to stop it.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.325Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":86},"message":"RUM endpoints enabled!","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2022-10-04T01:04:59.325Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":89},"message":"CORS related setting `apm-server.rum.allow_origins` allows all origins. Consider more restrictive setting for production use.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2022-10-04T01:04:59.325Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":102},"message":"Secret token is set, but SSL is not enabled.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.325Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":104},"message":"SSL disabled.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:04:59.735Z","log.logger":"request","log.origin":{"file.name":"middleware/log_middleware.go","file.line":61},"message":"request ok","service.name":"apm-server","url.original":"/","http.request.method":"GET","user_agent.original":"apm-agent-go/2.0.0 (filebeat)","source.address":"127.0.0.1","http.request.body.bytes":0,"http.request.id":"0eed0179-f6f5-416a-b9a3-d25501c4c78a","event.duration":140808,"http.response.status_code":200,"ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:01.449Z","log.logger":"request","log.origin":{"file.name":"middleware/log_middleware.go","file.line":61},"message":"request ok","service.name":"apm-server","url.original":"/","http.request.method":"GET","user_agent.original":"Go-http-client/1.1","source.address":"172.18.0.1","http.request.body.bytes":0,"http.request.id":"7ff24df3-48e8-475f-8c21-acbcbdb9d7bd","event.duration":129688,"http.response.status_code":200,"ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2022-10-04T01:05:09.238Z","log.logger":"beater.profiling","log.origin":{"file.name":"apm-server/main.go","file.line":308},"message":"failed to fetch cluster name from Elasticsearch: context deadline exceeded","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.247Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":348},"message":"registered profiling collection (technical preview)","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.247Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path / added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.247Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /config/v1/agents added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.247Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /config/v1/rum/agents added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /intake/v2/rum/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /intake/v3/rum/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /intake/v2/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /v1/traces added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /v1/metrics added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":133},"message":"Path /v1/logs added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"beater","log.origin":{"file.name":"beater/server.go","file.line":193},"message":"Starting apm-server [0a8e0ab7da29b7a617ad3bfa5c27ec1a29e3665e built 2022-10-03 12:41:43 +0200 +0200]. Hit CTRL-C to stop it.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":110},"message":"Stop listening on: :8200","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":86},"message":"RUM endpoints enabled!","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":89},"message":"CORS related setting `apm-server.rum.allow_origins` allows all origins. Consider more restrictive setting for production use.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":102},"message":"Secret token is set, but SSL is not enabled.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":104},"message":"SSL disabled.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.248Z","log.logger":"beater","log.origin":{"file.name":"beater/server.go","file.line":208},"message":"Server stopped","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.249Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":221},"message":"transaction metrics aggregation stopped","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.249Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":221},"message":"service destinations aggregation stopped","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2022-10-04T01:05:09.249Z","log.logger":"beater","log.origin":{"file.name":"beater/beater.go","file.line":368},"message":"context canceled","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-10-04T01:05:09.250Z","log.logger":"esclientleg","log.origin":{"file.name":"eslegclient/connection.go","file.line":108},"message":"elasticsearch url: [http://elasticsearch:9200","service.name":"apm-server","ecs.version":"1.6.0"}](http://elasticsearch:9200/)
--- FAIL: TestProfiling (27.92s)
```

--- a/systemtest/profiling_test.go
+++ b/systemtest/profiling_test.go
@@ -22,9 +22,11 @@ import (
 	"net/url"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/metadata"
 
 	"github.com/elastic/go-elasticsearch/v8/esapi"
@@ -99,7 +101,12 @@ func TestProfiling(t *testing.T) {
 		})
 
 	apmServerURL, _ := url.Parse(apmIntegration.URL)
-	conn, err := grpc.Dial(apmServerURL.Host, grpc.WithInsecure())
+
+	dialCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(dialCtx, apmServerURL.Host,
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()))
 	require.NoError(t, err)
 	defer conn.Close()
 	client := profiling.NewCollectionAgentClient(conn)

